### ISSUE TITLE
[BUG][3.6][New Storage] Pager related errors

### ISSUE THREAD

**Issue by JarJak** [bolt/bolt#7553]
Details
-------

 - Relevant Bolt Version: 3.6 with setcontent_legacy = false

Reproduction
------------

### Bug summary

In newest Bolt 3.6 (with the New Storage Layer by @rossriley ) setting setcontent_legacy to false breaks almost every page, while going back to true makes everything working.

### Specifics

Error I get is:

```
2/2
Twig_Error_Runtime in Template.php line 447:
An exception has been thrown during the rendering of a template ("Notice: Undefined variable: pager") in "partials/_header.twig" at line 6.

1/2
ContextErrorException in Environment.php(467) : eval()'d code line 57:
Notice: Undefined variable: pager
```

`partials/_header.twig` line 6 looks like this:

```
{% setcontent my_content = ct.slug where {slug: 'example-slug'} returnsingle %}
```

I don't use `$pager` or `pager()` anywhere in my code, however I've found out it is something used while rendering `{{ setcontent }}` in Twig here: https://github.com/bolt/bolt/blob/3.5/src/Twig/SetcontentNode.php#L71

### Steps to reproduce

On fresh Bolt 3.6 install:
1. in config.yml set setcontent_legacy to false
2. in default (theme-2018) record.twig template, add inside main section:
```
    {% set ct = config.get('contenttypes').entries %}
    {% setcontent my_content = ct.slug where {slug: 'example-slug'} returnsingle %}
```

### Expected result

200

### Actual result

500

![screen shot 2018-07-10 at 17 07 35](https://user-images.githubusercontent.com/4981490/42519398-44f6ac86-8464-11e8-8ca9-026755d7f144.png)

**Comment by JarJak** [comment]
OK I have tried it on Fresh Bolt 3.6 install and got 2 different errors.

First (unrelated):
![screen shot 2018-07-10 at 17 00 26](https://user-images.githubusercontent.com/4981490/42518722-cf1d4976-8462-11e8-9c84-03180d174da5.png)

line 5 is: https://github.com/bolt/themes/blob/master/base-2018/partials/_header.twig#L5

after commenting out that line I've got second error.

Second (probably related):
![screen shot 2018-07-10 at 17 07 35](https://user-images.githubusercontent.com/4981490/42519238-d5f21f82-8463-11e8-969f-20321b1c3f5a.png)

Error is different probably because in my custom theme I don't use pager at all.
I've updated steps to reproduce.
